File: src/main/java/com/example/cqrs/entity/read/write/event/base/AccountEvent.java
package com.example.cqrs.entity.read.write.event.base;

import java.time.LocalDateTime;

/**
 * 모든 이벤트는 AccountEvent 인터페이스를 구현하고, BaseAccountEvent 추상 클래스를 상속받습니다.
 */
public interface AccountEvent {

    String getAccountId();
    LocalDateTime getEventDate();

}

File: src/main/java/com/example/cqrs/command/entity/event/base/AccountEvent.kt
package com.example.cqrs.command.entity.event.base

import com.example.cqrs.command.entity.event.enumerate.EventStatus
import com.example.cqrs.command.entity.event.base.metadata.EventMetadata
import jakarta.persistence.Column
import jakarta.persistence.DiscriminatorValue
import jakarta.persistence.Entity
import java.time.LocalDateTime

/**
 * 계좌 관련 이벤트의 기본 클래스
 */
@DiscriminatorValue("AccountEvent")
@Entity
abstract class AccountEvent(
    id: Long? = null,

    @Column(nullable = false)
    val accountId: String,

    @Column(nullable = true)
    val amount: Double?,

    eventDate: LocalDateTime,
    metadata: EventMetadata,
    status: EventStatus = EventStatus.PENDING,
    version: Long? = null
) : BaseEvent(id, eventDate, metadata, status, version)
